.github/ISSUE_TEMPLATE/use-case.md

```diff
@@ -0,0 +1,44 @@
+---
+name: Use Case
+about: Use this template to propose an LWS use case.
+title: "[UC] <<brief description of use case>>"
+labels: triage, usecase
+assignees: ''
+
+---
+
+**As a [primary actor],**
+**I want [an action or feature],**
+**So that [a reason or benefit].**
+
+## Preconditions:
+
+*What preconditions do we expect there to be fulfilled prior to this use case being triggered?*
+
+## Trigger:
+
+*What triggers this use case?*
+
+## Actors:
+
+*Describe the primary actor, and any other relevant actors involved in this use case*
+
+## Distinction:
+
+*What is the challenge in realizing this use case? What distinguishes it from other, similar use cases?*
+
+## Scenario:
+
+*A more detailed happy-case scenario where this use case would be relevant*
+
+## Alternative case(s):
+
+*What alternative flows do we think a system must cover for this use case?*
+
```

Comment:
Happy, alternative, and error scenarios serve different purposes:

```suggestion

## Error scenario

*What unexpected issues or errors might arise, and how should the system handle them?*

```

Reply:
For me an error scenario could be seen as an alternative scenario, but happy to make this distinction if it can provide additional clarity for the use case description.

Reply:
I agree that the error scenario can be seen as an alternative scenario. I don't have a strong opinion on it but I do think the distinction is worthwhile. At the end, if this information can be explicitly described, it can be considered. Otherwise, it may or may not be taken into account when requirements are proposed and reviewed, and possibly codified. It will slip through the specification cracks ;)

The distinction I tried to make with happy and alternative was that they are successful - the latter being a legitimate path towards the goal - and well error is simply a fail.

strawperson use case: user tries to save their article in their storage. storage says nope because they don't have enough disk space. what an application/user to do?

(so, an alternative could hypothetically be that the user contacts their admin by email or ssh into their server to free up space or whatever, but that doesn't sound exciting. so what would the system need to do to handle the error and still keep the user (and their application for instance) in their flow, and continue - instructions to the app to inform the client and/or try quietly behind the scenes? have their app catch fire? send a structured message to the client ...

another: user tries to access a resource and server says that they're unauthorized. what happens next? we could tell the client that "we know you're unauthorized but if you really think you should have access to this, maybe you can contact us.." (see for instance client errors: https://github.com/solid/specification/pull/253#issuecomment-895062229 -- these are actual needs and implementation).

so, while happy/alternative are fine, but error to me says something a bit different and important.

Reply:
I'm confused by what "alternative use case" means, and slightly "error use case" (maybe still due to my lack of experience). Based on all the comments above, my understanding is like this:

- "Happy use case": The primary use case / flow of user interaction for making the use case happen, leading to an expected outcome;
- "Alternative use case": Other user interactions that will result in (?, or trigger) the same (?, or similar, in terms of the technical details of the expected outcome) use case / expected outcome (?, or technical details), in the same (?, or another) app;
- "Error use case": When executing the use case / interaction, what errors / exceptions may emerge, and (if known) how (?, or criteria on how) to deal with them.

Every question mark indicates a confusion, to the word / phrase before it; they also include a description of the alternatives (thus confusions) I see.